### PR TITLE
bt-290: Create badges and hard skills endpoints

### LINKED ISSUE #290: Create endpoint to get all badges and hard skills from db for select
Need to create GET endpoint for select options data (Screenshot 1, Screenshot 2)

- [x] create endpoint for hard skills
- [x] create endpoint for badges

(Screenshot 1)
Image

(Screenshot 2)
Image

### PR DESCRIPTION
**Pull request by Simpsonkk**

### CODE REVIEW

**Review by JosuerBague — CHANGES_REQUESTED** [review]
swagger annotation

**Comment by JosuerBague** [comment]
Tested both endpoints with dummy data in db, both endpoints return expected data.

Screenshot 2023-09-08 143028

Screenshot 2023-09-08 143334

File: backend/src/bundles/bsa-badges/bsa-badges.controller.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            method: 'GET',
+            handler: () => this.findAll(),
+        });
+    }
+
+    private async findAll(): Promise<ApiHandlerResponse> {
+        return {
+            status: HttpCode.OK,
+            payload: await this.bsaBadgesService.findAll(),
+        };
```
Comment (MaksymGavatuk) [inline backend/src/bundles/bsa-badges/bsa-badges.controller.ts]: As for me this variant better
```suggestion
        const badges = await this.bsaBadgesService.findAll(),
        
        return {
            status: HttpCode.OK,
            payload: badges
        };
```

File: backend/src/bundles/bsa-badges/bsa-badges.repository.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class BSABadgesRepository implements Repository {
+    private bsaBadgesModel: typeof BSABadgesModel;
+
+    public constructor(bsaBadgesModel: typeof BSABadgesModel) {
+        this.bsaBadgesModel = bsaBadgesModel;
+    }
+
+    public async findAll(): Promise<BSABadgeEntity[]> {
+        const badges = await this.bsaBadgesModel.query().execute();
+        return badges.map((it) => BSABadgeEntity.initialize(it));
```
Comment (MaksymGavatuk) [inline backend/src/bundles/bsa-badges/bsa-badges.repository.ts:15]: You can use just:
```suggestion
        return badges.map(BSABadgeEntity.initialize);
```

File: backend/src/bundles/hard-skills/hard-skills.controller.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            method: 'GET',
+            handler: () => this.findAll(),
+        });
+    }
+
+    private async findAll(): Promise<ApiHandlerResponse> {
+        return {
+            status: HttpCode.OK,
+            payload: await this.hardSkillsService.findAll(),
+        };
```
Comment (MaksymGavatuk) [inline backend/src/bundles/hard-skills/hard-skills.controller.ts]: The same

File: backend/src/bundles/hard-skills/hard-skills.model.ts [hunk 4]
```diff
@@ -5,27 +5,22 @@ import {
...
             join: {
                 from: `${DatabaseTableName.HARD_SKILLS}.${HardSkillsTableColumn.ID}`,
-                through: {
-                    from: `${DatabaseTableName.TALENT_HARD_SKILLS}.${TalentHardSkillsTableColumn.HARD_SKILL_ID}`,
-                    to: `${DatabaseTableName.TALENT_HARD_SKILLS}.${TalentHardSkillsTableColumn.USER_DETAILS_ID}`,
-                },
-                to: `${DatabaseTableName.USER_DETAILS}.${UserDetailsTableColumn.ID}`,
+                to: `${DatabaseTableName.TALENT_HARD_SKILLS}.${TalentHardSkillsTableColumn.HARD_SKILL_ID}`,
             },
         },
```
Comment (MaksymGavatuk) [inline backend/src/bundles/hard-skills/hard-skills.model.ts]: I think you don't need this relation

File: backend/src/bundles/hard-skills/hard-skills.repository.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+class HardSkillsRepository implements Repository {
+    private hardSkillsModel: typeof HardSkillsModel;
+
+    public constructor(hardSkillsModel: typeof HardSkillsModel) {
+        this.hardSkillsModel = hardSkillsModel;
+    }
+
+    public async findAll(): Promise<HardSkillsEntity[]> {
+        const skills = await this.hardSkillsModel.query().execute();
+        return skills.map((it) => HardSkillsEntity.initialize(it));
```
Comment (MaksymGavatuk) [inline backend/src/bundles/hard-skills/hard-skills.repository.ts:15]:
```suggestion
        return skills.map(HardSkillsEntity.initialize);
```

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

File: backend/src/bundles/bsa-badges/bsa-badges.repository.ts (hunk 2 above)
Comment (Simpsonkk) [reply backend/src/bundles/bsa-badges/bsa-badges.repository.ts:15]: Do not pass function directly to `.map(…)`. eslint 
image

File: backend/src/bundles/hard-skills/hard-skills.repository.ts (hunk 5 above)
Comment (Simpsonkk) [reply backend/src/bundles/hard-skills/hard-skills.repository.ts:15]: Do not pass function directly to `.map(…)`.eslint
image

File: backend/src/bundles/bsa-badges/bsa-badges.controller.ts (hunk 1 above)
Comment (Simpsonkk) [reply backend/src/bundles/bsa-badges/bsa-badges.controller.ts]: I just don`t like extra variables :)
I added new variable

File: backend/src/bundles/hard-skills/hard-skills.controller.ts (hunk 3 above)
Comment (Simpsonkk) [reply backend/src/bundles/hard-skills/hard-skills.controller.ts]: done

File: backend/src/bundles/hard-skills/hard-skills.model.ts (hunk 4 above)
Comment (Simpsonkk) [reply backend/src/bundles/hard-skills/hard-skills.model.ts]: I added it because I saw the relation in migration.
Done.

**Review by anton-otroshchenko — APPROVED** [review]

**Review by JosuerBague — APPROVED** [review]

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
we should create seeds to create base badges and skills.
https://knexjs.org/guide/migrations.html#seed-files

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**